pkg/virt-launcher/virtwrap/sriov/hostdev.go

```diff
@@ -98,3 +102,82 @@ func decoratePciAddressField(addressField string) (*api.Address, error) {
 	}
 	return decoratedAddrField, nil
 }
+
+type deviceDetacher interface {
+	DetachDevice(xmlData string) error
+}
+
+func DetachHostDevices(dom deviceDetacher, hostDevices []api.HostDevice) error {
+	for _, device := range hostDevices {
+		devXML, err := xml.Marshal(device)
+		if err != nil {
+			return fmt.Errorf("failed to encode (xml) device %v, err: %v", device, err)
+		}
+		err = dom.DetachDevice(string(devXML))
+		if err != nil {
+			return fmt.Errorf("failed to detach device %s, err: %v", devXML, err)
+		}
+		log.Log.Infof("Live migration, hot-unplug device: %s (%v)", device.Alias.Name, device.Source.Address)
+	}
+	return nil
+}
+
+func FilterHostDevices(domainSpec *api.DomainSpec) []api.HostDevice {
+	var hostDevices []api.HostDevice
+
+	for _, device := range domainSpec.Devices.HostDevices {
+		if device.Alias != nil && strings.HasPrefix(device.Alias.Name, AliasPrefix) {
+			hostDevices = append(hostDevices, device)
+		}
+	}
+	return hostDevices
+}
+
+type Registrar interface {
+	Register(callback interface{}) error
+	Deregister() error
+}
+
+func WaitHostDevicesToDetach(detacher Registrar, hostDevices []api.HostDevice, timeout time.Duration, f func() error) error {
```

Comment:
Why the do we need to pass the `sriov.DetachHostDevices` function as argument instead of using it here,
I mean the function name is `WaitHostDevicesToDetach` its not that we going to use it for somthing else right?
If the intention is to reuse this function then I think the name and log message should be more generic.

In any case please change the name of `f` to something more informative